src/ast.c

```diff
@@ -267,6 +268,7 @@ static jl_ast_context_t *jl_ast_ctx_enter(void)
 
 static void jl_ast_ctx_leave(jl_ast_context_t *ctx)
 {
+    JL_SIGATOMIC_END();
```

Comment:
something like that. might need to be unix only, would need to run it through CI first